### PR TITLE
Subscribers: Update Odyssey stats links to JP Cloud

### PR DESCRIPTION
**Pull request by lupus2k**
Related to https://github.com/Automattic/dotcom-forge/issues/6364

## Proposed Changes

Update `View details` / `View all` link on Stats > Subscribers (tab) to Jetpack Cloud Subscribers for Atomic & Self hosted.
This also should update on Calypso and wp-admin Odyssey stats. 

image

## Testing Instructions

* Go to `/apps/odyssey-stats`
* Run `NODE_ENV=production yarn dev --sync` to sync the changes to your sandbox
* Sandbox `widgets.wp.com`
* Use a site with >0 subscribers
* On Calypso, go to `/stats/subscribers/:SITE_SLUG`
* On wp-admin, go to `/wp-admin/admin.php?page=stats#!/stats/subscribers/:SITE_SLUG`
* The `View details` at the bottom of the page should link to Jetpack Cloud subscribers

### CODE REVIEW

**Review by candy02058912 — APPROVED** [review]
✅ Atomic site
✅ Self-hosted site

I noticed that it will still be broken for Simple Classic, so created another issue to track that https://github.com/Automattic/dotcom-forge/issues/6466

File: client/my-sites/stats/stats-followers/index.jsx [hunk 1]
```diff
@@ -77,12 +81,17 @@ class StatModuleFollowers extends Component {
 		const summaryPageSlug = siteSlug || '';
 		// email-followers is no longer available, so fallback to the new subscribers URL.
 		// Old, non-functional path: '/people/email-followers/' + summaryPageSlug.
-		let summaryPageLink = '/people/subscribers/' + summaryPageSlug;
+		// If the site is Atomic or Jetpack self-hosted, it links to Jetpack cloud.
+		let summaryPageLink =
+			isAtomic || isJetpack
+				? `https://cloud.jetpack.com/subscribers/${ summaryPageSlug }`
+				: `/people/subscribers/${ summaryPageSlug }/${ isAtomic }/${ isJetpack }`;
```
Comment (candy02058912) [inline client/my-sites/stats/stats-followers/index.jsx:88]: This looks a bit weird, boolean values in the URL 🤔

Comment (candy02058912) [reply client/my-sites/stats/stats-followers/index.jsx:88]: Ah noticed this broke the Calypso Stats link for simple sites, it became `https://wordpress.com/people/subscribers/site.wordpress.com/false/false`, I created a followup PR: https://github.com/Automattic/wp-calypso/pull/89311

Comment (lupus2k) [reply client/my-sites/stats/stats-followers/index.jsx:88]: Thanks for the fix :tada:

**Pull request merged**